### PR TITLE
feat(server): shell fallback + safe.directory injection with tests

### LINKED ISSUE #892: Workspace provisioning fails on Windows: spawn /bin/sh ENOENT
## Description

When using `git_worktree` execution workspace policy with a `provisionCommand`, the workspace runtime fails on Windows with:

```
spawn /bin/sh ENOENT
```

## Root Cause

In `server/src/services/workspace-runtime.ts`, the shell fallback is:

```typescript
const shell = process.env.SHELL?.trim() || "/bin/sh";
```

On Windows, `process.env.SHELL` is typically unset, so it falls back to `/bin/sh` which does not exist. This affects two call sites (workspace provision commands and runtime service spawning).

## Fix

Fall back to `"sh"` on Windows (available via Git Bash on PATH):

```typescript
const shell = process.env.SHELL?.trim() || (process.platform === "win32" ? "sh" : "/bin/sh");
```

## Environment

- OS: Windows 11
- Shell: Git Bash
- paperclipai: running from source (master)
- Workspace policy: `git_worktree` with `provisionCommand: "pnpm install"`

### LINKED ISSUE #1684: Git 'dubious ownership' error on WSL/cross-OS workspace mounts
## Problem

When Paperclip creates git worktrees or runs git commands in isolated workspaces on WSL mounts (or any cross-OS mount where directory ownership differs), Git 2.35.2+ rejects the repo with a fatal "dubious ownership" error. This causes \`adapter_failed\` during workspace realization.

## Expected behavior

Git operations in workspace directories should succeed regardless of ownership differences, without modifying the user's global git config.

## Suggested approach

Inject \`safe.directory\` trust via \`GIT_CONFIG_COUNT\`/\`GIT_CONFIG_KEY_N\`/\`GIT_CONFIG_VALUE_N\` env vars on the child process, scoped to the specific workspace \`cwd\`.

### PR DESCRIPTION
**Pull request by octasoft-ltd**
## Summary

- **Shell fallback (PR #891 coverage):** Extracts `resolveShell()` helper that uses `$SHELL` when set, falls back to `sh` on Windows (Git Bash PATH lookup) or `/bin/sh` on Unix. Fixes ENOENT errors on Windows where `/bin/sh` does not exist.
- **safe.directory injection (PR #908 coverage):** Extracts `buildSafeDirectoryEnv()` helper that injects `GIT_CONFIG_COUNT`/`GIT_CONFIG_KEY_N`/`GIT_CONFIG_VALUE_N` env vars scoped to the workspace `cwd`, avoiding "dubious ownership" fatal errors on cross-OS mounts (e.g. WSL <-> Windows).
- Adds 13 unit tests covering shell resolution edge cases and safe.directory env construction.

Resolves #892, resolves #1684.

## Test plan

- [x] 7 tests for `resolveShell()`: SHELL set, trimmed, empty, whitespace-only, linux/darwin/win32 fallbacks
- [x] 6 tests for `buildSafeDirectoryEnv()`: fresh count, incremental, non-numeric, negative, scoped cwd, no process.env mutation
- [x] All 17 tests pass (4 existing + 13 new)
- [x] TypeScript typecheck passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by octasoft-ltd** [comment]
@greptileai please review the full PR changes after rebase

File: server/src/__tests__/workspace-runtime.test.ts [hunk 1]
```diff
@@ -875,6 +877,131 @@ describe("ensureRuntimeServicesForRun", () => {
...
+    const env = buildSafeDirectoryEnv("/specific/path");
+    expect(env.GIT_CONFIG_VALUE_0).toBe("/specific/path");
+    expect(env.GIT_CONFIG_VALUE_0).not.toBe("*");
+  });
+
+  it("does not mutate process.env", () => {
+    delete process.env.GIT_CONFIG_COUNT;
+    buildSafeDirectoryEnv("/tmp/test");
+    expect(process.env.GIT_CONFIG_COUNT).toBeUndefined();
+  });
```
[the remark this answers is not in the document]
Comment (octasoft-ltd) [reply server/src/__tests__/workspace-runtime.test.ts:1006]: Fixed in 7c5ce3a6 — added assertions for `GIT_CONFIG_KEY_0` and `GIT_CONFIG_VALUE_0` to complete the non-mutation guard.

File: server/src/__tests__/workspace-runtime.test.ts [hunk 2]
```diff
@@ -4,11 +4,14 @@ import os from "node:os";
 import path from "node:path";
 import { promisify } from "node:util";
 import { afterEach, describe, expect, it } from "vitest";
```
[the remark this answers is not in the document]
Comment (octasoft-ltd) [reply server/src/__tests__/workspace-runtime.test.ts:6]: All Greptile comments addressed in e14c5e4a: removed unused `vi` import, refactored `parseInt` guard to use `Number.isFinite`, and replaced unsafe type cast with `Object.entries().filter()` for proper undefined handling.
